src/sentry/models/widget.py

```diff
@@ -0,0 +1,92 @@
+from __future__ import absolute_import, print_function
+
+from django.db import models
+from django.utils import timezone
+from jsonfield import JSONField
+
+from sentry.constants import ObjectStatus
+from sentry.db.models import BoundedPositiveIntegerField, FlexibleForeignKey, Model, sane_repr
+
+
+class WidgetDisplayTypes(object):
+    LINE_CHART = 0
+    AREA_CHART = 1
+    STACKED_AREA_CHART = 2
+    BAR_CHART = 3
+    PIE_CHART = 4
+    TABLE = 5
+    WORLD_MAP = 6
+    PERCENTAGE_AREA_CHART = 7
+
+    @classmethod
+    def as_choices(cls):
+        return [
+            (cls.LINE_CHART, 'line'),
+            (cls.AREA_CHART, 'area'),
+            (cls.STACKED_AREA_CHART, 'stacked_area'),
+            (cls.BAR_CHART, 'bar'),
+            (cls.PIE_CHART, 'pie'),
+            (cls.TABLE, 'table'),
+            (cls.WORLD_MAP, 'world_map'),
+            (cls.PERCENTAGE_AREA_CHART, 'percentage_area_chart')
+        ]
+
+
+class WidgetDataSourceTypes(object):
+    DISCOVER_SAVED_SEARCH = 0
+
+    @classmethod
+    def as_choices(cls):
+        return [
+            (cls.DISCOVER_SAVED_SEARCH, 'discover_saved_search'),
+        ]
+
+
+class WidgetDataSource(Model):
+    """
+    A dashboard widget.
+    """
+    __core__ = True
+
+    widget = FlexibleForeignKey('sentry.Widget')
+    type = BoundedPositiveIntegerField(choices=WidgetDataSourceTypes.as_choices())
+    name = models.CharField(max_length=255)
+    data = JSONField(default={})  # i.e. saved discover query
```

Comment:
What do we not know about what is being stored in this field? Is there a way we could get some clarity on that before going with a JSON field?

Reply:
Yes so @markstory for right now it's going to be a saved discover query. Take a look at https://github.com/getsentry/sentry/tree/feat/dashboard/static-overview-dashboard/src/sentry/static/sentry/app/views/organizationDashboard/data/queries for examples.